src/PostgREST/RangeQuery.hs

```diff
@@ -38,12 +41,13 @@ rangeParse range = do
       rangeIntersection lower upper
     Nothing -> rangeGeq 0
 
-rangeRequested :: RequestHeaders -> NonnegRange
-rangeRequested = rangeParse . fromMaybe "" . lookup hRange
+rangeRequested :: RequestHeaders -> Maybe NonnegRange
```

Comment:
You could modify that function to also operate without a Maybe:

```haskell
setTopLevelRange :: NonnegRange -> M.HashMap String NonnegRange -> M.HashMap String NonnegRange
setTopLevelRange r ranges
  | r == rangeGeq 0 = ranges
  | otherwise       = M.insert "limit" r ranges
```